pydrive2/test/test_file.py

```diff
@@ -356,6 +402,128 @@ def test_Files_Get_Content_Buffer(self):
 
         self.DeleteUploadedFiles(drive, [file1["id"]])
 
+    def test_Files_Get_Content_Buffer_resourceKey_missing(self):
+        """404 expected for file secured with resourceKey when not provided."""
+
+        ga = auth_with_resource_key_mock()
+
+        drive = GoogleDrive(ga)
+        file1 = drive.CreateFile(
+            {
+                "id": "0BxphPoRgwhnodHNjS3JESnFNS1E",
+            }
+        )
+        with self.assertRaisesRegex(
+            ApiRequestError, "HttpError 404 when requesting"
+        ):
+            pydrive_retry(file1.GetContentIOBuffer)
+
+    def test_Files_Get_Content_Buffer_resourceKey(self):
+        """End to end scenario with real file."""
+        ga = auth_with_resource_key_mock()
+        drive = GoogleDrive(ga)
+        file1 = drive.CreateFile(
+            {
+                "id": "0BxphPoRgwhnodHNjS3JESnFNS1E",
+                "resourceKey": "0-vjzOveuin3fnf4LUlfsD3A",
+            }
+        )
+
+        buffer1 = pydrive_retry(file1.GetContentIOBuffer)
+
+        self.assertEqual(len(buffer1), 12)
+
+    @pytest.mark.manual
+    def test_Files_Get_Content_Buffer_resourceKey_missing_real(self):
+        """
+        404 expected for file secured with resourceKey when not provided.
+        End to end scenario with real public file.
+        """
+        drive = GoogleDrive(self.ga)
+        file1 = drive.CreateFile(
+            {
+                "id": "0BxphPoRgwhnodHNjS3JESnFNS1E",
+            }
+        )
+        with self.assertRaisesRegex(
+            ApiRequestError, "HttpError 404 when requesting"
+        ):
+            pydrive_retry(file1.GetContentIOBuffer)
+
+    @pytest.mark.manual
+    def test_Files_Get_Content_Buffer_resourceKey_real(self):
+        """End to end scenario with real public file."""
+        drive = GoogleDrive(self.ga)
+        file1 = drive.CreateFile(
+            {
+                "id": "0BxphPoRgwhnodHNjS3JESnFNS1E",
+                "resourceKey": "0-vjzOveuin3fnf4LUlfsD3A",
+            }
+        )
+
+        buffer1 = pydrive_retry(file1.GetContentIOBuffer)
+
+        self.assertEqual(len(buffer1), 6128902)
+
+    def test_Files_Get_Content_File_resourceKey_missing(self):
```

Comment:
do we really need both Content_File _and_ Content_Buffer?

Reply:
yep, it's just not exactly clear how we draw the line - it seems all APIs (delete, get content, etc) require this key, right?

may be what we should do instead is to test, just an idea ... I think it's fine as it is and we can move forward. Since the whole test suite is not perfect here.

- if key is passed then it's included automatically into the API calls (test for one of the wrappers)?
- test one of the calls E2E
- if key is not passed but required - test 404

something like this. Otherwise it's not obvious (to me) why we specifically pick on this subset of APIs.

Reply:
Actually I only required GetContentIOBuffer for my own project, so started with it. I've also done GetContentFile and FetchMetadata because I found them similar enough and easy to implement. I stopped there because I didn't want to go into a broader change (which I selfishly don't need :) ), as I feel implementing it "properly" (meaning in a nice way) in all methods would require a deeper refactoring.
That's why I flagged my PR as only partial implementation.

Reply:
yeah, my question was primarily about the number of tests overall ... I don't feel we should be covering this for each individual API call - it would be too much. Anyways, probably can keep it this way for now.